dissolve-struct-lib/src/main/scala/ch/ethz/dalab/dissolve/classification/StructSVMModel.scala

```diff
@@ -31,11 +31,42 @@ class StructSVMModel[X, Y](
     dissolveFunctions: DissolveFunctions[X, Y]) =
     this(weights, ell, ellMat, dissolveFunctions, -1)
 
+  def addToWeight(delta:Vector[Double]): Unit = {
+    //if delta is sparse we can speedup the addition even more
+     delta match{
+       case delta:SparseVector[Double] => 
+         val indexes = delta.index
+         for(i <- 0 until indexes.length){
+           val idx = indexes(i)
+           weights(idx) += delta.data(i)
+         }
+       case _ => 
+         for(i <- 0 until delta.length){
+           weights(i) += delta(i)
+         }
+     }
+  }
+  
+  def subtractFromWeight(delta:Vector[Double]): Unit = {
+     delta match{
+       case delta:SparseVector[Double] => 
+         val indexes = delta.index
+         for(i <- 0 until indexes.length){
+           val idx = indexes(i)
+           weights(idx) -= delta.data(i)
+         }
+       case _ => 
+         for(i <- 0 until delta.length){
+           weights(i) -= delta(i)
+         }
+     }
+  }
+    
```

Comment:
I think the implementations in breeze create a new vector object. Meaning that a+b would create a new dense-vector object. I did not see an implementation in breeze in which a+b would add b to the vector a, thus manipulating a, and not create a new object